components/system-broker/pkg/oauth/token_provider.go

```diff
@@ -48,134 +35,13 @@ const (
-		secretNamespace: config.SecretNamespace,
-	}
-}
-
-func (c *OAuthTokenProvider) GetAuthorizationToken(ctx context.Context) (httputils.Token, error) {
-	credentials, err := c.extractOAuthClientFromSecret(ctx)
-	if err != nil {
-		return httputils.Token{}, errors.Wrap(err, "while get credentials from secret")
-	}
-
-	return c.getAuthorizationToken(ctx, credentials)
-}
-
-func (c *OAuthTokenProvider) WaitForCredentials(ctx context.Context) error {
-	err := wait.Poll(time.Second, time.Minute*3, func() (bool, error) {
-		secret := &v1.Secret{}
-		err := c.k8sClient.Get(ctx, client.ObjectKey{
-			Namespace: c.secretNamespace,
-			Name:      c.secretName,
-		}, secret)
-		// it fails on connection-refused error on first call and it restarts our application.
-		if err != nil {
-			log.C(ctx).Warnf("secret %s not found", c.secretName)
-			return false, nil
-		}
-		return true, nil
-	})
-
-	return errors.Wrapf(err, "while waiting for secret %s", c.secretName)
-}
-
-func (c *OAuthTokenProvider) extractOAuthClientFromSecret(ctx context.Context) (credentials, error) {
-	secret := &v1.Secret{}
-	err := c.k8sClient.Get(ctx, client.ObjectKey{
-		Namespace: c.secretNamespace,
-		Name:      c.secretName,
-	}, secret)
-	if err != nil {
-		return credentials{}, err
-	}
-
-	return credentials{
-		clientID:       string(secret.Data[clientIDKey]),
-		clientSecret:   string(secret.Data[clientSecretKey]),
-		tokensEndpoint: string(secret.Data[tokensEndpointKey]),
-	}, nil
-}
-
-func (c *OAuthTokenProvider) getAuthorizationToken(ctx context.Context, credentials credentials) (httputils.Token, error) {
-	log.C(ctx).Infof("Getting authorization token from endpoint: %s", credentials.tokensEndpoint)
-
-	form := url.Values{}
-	form.Add(grantTypeFieldName, credentialsGrantType)
-	form.Add(scopeFieldName, scopes)
-	body := strings.NewReader(form.Encode())
-	request, err := http.NewRequest(http.MethodPost, credentials.tokensEndpoint, body)
-	if err != nil {
-		return httputils.Token{}, errors.Wrap(err, "Failed to create authorisation token request")
+func NewTokenProvider(config *Config, httpClient httputils.Client) (httputils.TokenProvider, error) {
+	if config.Local {
+		return NewTokenProviderFromValue(config.TokenValue), nil
 	}
-
-	//we can use a request provider or maybe its an overkill (reason for making it was correlation ids but then i moved them to a transport)
-	//input := httputils.RequestInput{
-	//	Method:  http.MethodPost,
-	//	URL:     credentials.tokensEndpoint,
-	//	Body:    body,
-	//	Headers: headers,
-	//}
-	//
-	//log.C(ctx).Errorf("%+v", input)
-	//request, err := c.requestProvider.Provide(ctx, input)
-	//if err != nil {
-	//	return httputils.Token{}, errors.Wrap(err, "while creating authorisation token request")
-	//}
-
-	request.SetBasicAuth(credentials.clientID, credentials.clientSecret)
-	request.Header.Set(contentTypeHeader, contentTypeApplicationURLEncoded)
-
-	response, err := c.httpClient.Do(request)
+	k8sClient, err := prepareK8sClient()
```

Comment:
It'sa little weird to me that a `/pkg/oauth` package with a public function with the following signiture:

```
func NewTokenProvider(config *Config, httpClient httputils.Client) (httputils.TokenProvider, error)
```

would internally have some k8s specific logic, but since this is for the sake of not having a gigantic `main.go` and I cannot give you a much better proposal for where to move this and have it as dependency, I guess we can live with it.

Reply:
Also it is used to unify the oauth provider with k8s secret credentials and a local provider which just returns a predefined value. This way when we rely on a single interface we can manipulate it from outside and have the 
`--oauth_provider.local` flag to easily run it locally.